src/transformers/modeling_tf_t5.py

```diff
@@ -783,34 +784,6 @@ def dummy_inputs(self):
         }
         return dummy_inputs
 
-    def _shift_right(self, input_ids):
```

Comment:
The move of `_shift_right` is inconsistent with the takeaways from the survey to have individual self-contained files for the models at the price of duplicate code.

Reply:
I think @thomwolf 's survey takeaways were a little more nuanced. What I gathered was
(1) whether code-sharing is misleading/useful should be decided on a case by case basis.
(2) Code-sharing is mostly misleading when it involves the core deeplearning ops of the forward pass.
(3) code sharing is mostly useful when it involves boilerplate/data-processing code.

That's why I approved the qa_loss refactor PR [#6430](https://github.com/huggingface/transformers/pull/6430)  -- I didn't think computing QA loss was "core modeling/deep learning" code, but closer to plumbing.

But I defer to the boss, they are his takeaways after all.

Reply:
Also why I approved the qa_loss refactor PR [#6430](https://github.com/huggingface/transformers/pull/6430)  -- I didn't think computing QA loss was "core modeling/deep learning" code, but closer to plumbing.

Here, I decided shifting tokens right while ignoring pad tokens was not core deep learning code.